src/v/config/configuration.cc

```diff
@@ -926,8 +926,9 @@ configuration::configuration()
       *this,
       "tx_log_stats_interval_s",
       "How often to log per partition tx stats, works only with debug logging "
-      "enabled.",
-      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      "enabled. (deprecated starting v24.2)",
```

Comment:
Does this PR also tags this property as deprecated programmatically?

Reply:
@Deflaimun what do you mean by "deprecated programatically"?  I marked it as deprecated in the configuration settings, is there more to it?  Either way I removed the note from the description.

Reply:
Looks like there's two ways to tag a property as deprecated. 
1. add it on `configuration.h` https://github.com/redpanda-data/redpanda/blob/dev/src/v/config/configuration.h#L69
2.  put the visibility as deprecated

I wasn't aware of 2. 
But now, I don't know why we have two ways of doing the same thing.. 

Hmm anyway, the reasoning was that we extract this from the property automation, so I was just making sure that we're capturing the change too.